BiometricLockerTests/BiometricLockerTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  BiometricLockerTests.swift
+//  BiometricLockerTests
+//
+//  Created by Igor Ranieri on 16.04.18.
+//  Copyright © 2018 Bakken&Bæck. All rights reserved.
+//
+
+import XCTest
+import LocalAuthentication
+@testable import BiometricLocker
+
+class BiometricLockerTests: XCTestCase {
+
+    override func setUp() {
+        super.setUp()
+
+        // Since lockers are not namespaced or anything, if you have multiple lockers, they'll all use the same UserDefaults date value.
+        // To ensure we're always starting from scratch, create one and unlock to clear the UserDefaults value. Will improve on this.
+        BiometricLocker(localizedReason: "").unlock()
+    }
+
+    func testLockUnlock() {
+        let expectation = self.expectation(description: "Locking")
+
+        let locker = BiometricLocker(localizedReason: "", withUnlockedTimeAllowance: 3)
+        XCTAssertFalse(locker.isLocked)
+
+        locker.lock()
+        XCTAssertFalse(locker.isLocked)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 4) {
+            XCTAssertTrue(locker.isLocked)
+            expectation.fulfill()
+        }
+
+        self.wait(for: [expectation], timeout: 5)
+    }
+
+    func testLockUnlockImmediatelly() {
+        let locker = BiometricLocker(localizedReason: "")
+        XCTAssertFalse(locker.isLocked)
+
+        locker.lock(.now)
+        XCTAssertTrue(locker.isLocked)
+
+        locker.unlock()
+        XCTAssertFalse(locker.isLocked)
+    }
+
+    func testLockingAfterTimeAllowance() {
+        let locker = BiometricLocker(localizedReason: "")
+        XCTAssertFalse(locker.isLocked)
+        let expectation = self.expectation(description: "Allowance time lock")
+
+        locker.lock(.now)
+        XCTAssertTrue(locker.isLocked)
+
+        locker.unlock()
+        XCTAssertFalse(locker.isLocked)
+
+        let otherLocker = BiometricLocker(localizedReason: "", withUnlockedTimeAllowance: 3)
+        otherLocker.lock(.afterTimeAllowance)
+        XCTAssertFalse(otherLocker.isLocked)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 6) {
+            XCTAssertTrue(otherLocker.isLocked)
+            expectation.fulfill()
+        }
+
+        self.wait(for: [expectation], timeout: 8)
+    }
+
+    func testLockingAfterCustomInterval() {
+        let locker = BiometricLocker(localizedReason: "")
+        XCTAssertFalse(locker.isLocked)
+        let expectation = self.expectation(description: "Custom time lock")
+
+        locker.lock(.now)
+        XCTAssertTrue(locker.isLocked)
+
+        locker.unlock()
+        XCTAssertFalse(locker.isLocked)
+
+        // We set the time a specific allowance, just to be sure that it's being ignored!
+        let otherLocker = BiometricLocker(localizedReason: "", withUnlockedTimeAllowance: 3)
+        otherLocker.lock(.afterTimeInterval(8))
+        XCTAssertFalse(otherLocker.isLocked)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 6) {
```

Comment:
I don't know why my stuff keeps getting resolved here but I still think the math here is off unless I'm misunderstanding something

Reply:
Shouldn't this lock after 5 seconds, since interval is 8 and time allowance is 3?

Reply:
` otherLocker.lock(.afterTimeInterval(8))`, it should lock after 8s. That's the custom time interval. 👀

Reply:
```swift
 case .afterTimeInterval(let interval):
            date = Date(timeIntervalSinceNow: (interval - self.unlockedTimeAllowance))
``` 

What am I missing here?

Reply:
```swift
             let appWasInBackgroundForLongerThan5Minutes = Date().timeIntervalSince(applicationDidEnterBackgroundDate) > self.unlockedTimeAllowance
```
This is the check in `isLocked`. We subtract it here to offset it there.

Reply:
ahaaaaa. That's not super-obvious. Can you please throw a comment in about that?

Reply:
[addressed]